src/components/atoms/Icon/Icon.js

```diff
@@ -34,9 +58,22 @@ export default class Icon extends HTMLElement {
       default:
         break;
     }
-    const iconContainer = document.createElement('span');
-    iconContainer.innerHTML = this.getIcon(icon, size);
-    this.shadowRoot.appendChild(iconContainer);
+
+    const iconElement = document.createElement('span');
+    iconElement.innerHTML = this.getIcon(icon, size);
+    this.shadowRoot.appendChild(iconElement);
+
+    // Append the icon element to the container
+    container.appendChild(iconElement);
```

Comment:
Here we're appending the same icon to two different locations in the DOM. That latter (line 67) will overwrite the first, so there's no need to have line 64 here.